mllib/src/main/scala/org/apache/spark/mllib/recommendation/MatrixFactorizationModel.scala

```diff
@@ -199,12 +199,12 @@ object MatrixFactorizationModel extends Loader[MatrixFactorizationModel] {
       assert(formatVersion == thisFormatVersion)
       val rank = (metadata \ "rank").extract[Int]
       val userFeatures = sqlContext.parquetFile(userPath(path))
-        .map { case Row(id: Int, features: Seq[Double]) =>
-          (id, features.toArray)
+        .map { case Row(id: Int, features: Seq[_]) =>
+          (id, features.asInstanceOf[Seq[Double]].toArray)
```

Comment:
Strangely, this is how the scala compiler wanted it. It doesn't like matching on a type with generics, since they are erased.